### PR TITLE
[DOCS] Query strings are normalized for fuzzy (`~`) operator

### LINKED ISSUE #73299: Docu Change: Query String Queries, fuzziness queries not analyzed, missing documentation
The documentation for the fuzziness parameter in the query string query is missing the information that fuzziness queries are not analyzed.
This can lead to some confusion and unexpected results:
https://discuss.elastic.co/t/terms-with-fuzzy-operator-dont-use-mapping-analyzer-but-always-standard-analyzer/68593/4
Especially because the multi match queries which has a similar fuzziness setting behaves differently. 

There was one old documentation that was mentioning this:
```lowercase_expanded_terms      
Whether terms of wildcard, prefix, fuzzy, and range queries are to be automatically lower-cased or not (since they are not analyzed). Defaults to true.
```
https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-query-string-query.html

But the setting war removed and this documentation with it.

I propose the following change to the documentation of the Query String Query:

Change:
```
fuzziness
(Optional, string) Maximum edit distance allowed for matching. See Fuzziness for valid values and more information.
```
To:
```
fuzziness
(Optional, string) Maximum edit distance allowed for matching. See Fuzziness for valid values and more information. 
Fuzzy queries are lowercased and not analyzed.
```

### PR DESCRIPTION
**Pull request by jrodewig**
Notes that `fuzzy` queries made using the query string query's `~`
operator are normalized.

Closes #73299

### Preview
https://elasticsearch_73921.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-dsl-query-string-query.html#query-string-fuzziness

### CODE REVIEW

File: docs/reference/query-dsl/query-string-syntax.asciidoc [hunk 1]
```diff
@@ -116,14 +116,12 @@ Use with caution!
...
-We can search for terms that are
-similar to, but not exactly like our search terms, using the ``fuzzy''
-operator:
+You can run <<query-dsl-fuzzy-query,`fuzzy` queries>> using the `~` operator:
 
     quikc~ brwn~ foks~
 
-This uses the
-{wikipedia}/Damerau-Levenshtein_distance[Damerau-Levenshtein distance]
+For these queries, the query string is not <<analysis,analyzed>>. Instead, the
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/query-string-syntax.asciidoc:123]: For fuzzy queries, we will  [normalize](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-normalizers.html#analysis-normalizers) a query string. This means, that we will take an analyzer for the field or the provided analyzer,  take only token filters that can normalize and run the query string through it.

**Review by mayya-sharipova — COMMENTED** [review]
@jrodewig Sorry for a late review, I've just noticed your PR. I've left a comment.

File: docs/reference/query-dsl/query-string-syntax.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/query-string-syntax.asciidoc]: Thanks for the correction @mayya-sharipova. I pushed a few commits to fix this.

**Review by mayya-sharipova — APPROVED** [review]
@jrodewig Thanks James, the changes LGTM!

**Comment by jrodewig** [comment]
Thanks @mayya-sharipova!

**Pull request merged**